apps/gpc-prover-client/app/util/generateProof.ts

```diff
@@ -0,0 +1,107 @@
+import { Dispatch } from "react";
+import { POD } from "@pcd/pod";
+import {
+  GPCProofInputs,
+  deserializeGPCProofConfig,
+  gpcArtifactDownloadURL,
+  gpcProve,
+  serializeGPCBoundConfig,
+  serializeGPCRevealedClaims
+} from "@pcd/gpc";
+import { Identity } from "@semaphore-protocol/identity";
+
+export type ProofResult = {
+  config: string;
+  proof: string;
+  claims: string;
+};
+
+export const generateProof = async (
+  identity: Identity,
+  serializedIDPOD: string,
+  serializedPaystubPOD: string,
+  serializedProofRequest: string,
+  setProofResult: Dispatch<ProofResult>
+) => {
+  try {
+    if (!identity) {
+      throw new Error("Identity field cannot be empty!");
+    }
+
+    if (!serializedIDPOD) {
+      throw new Error("ID POD field cannot be empty!");
+    }
+
+    if (!serializedPaystubPOD) {
+      throw new Error("Paystub POD field cannot be empty!");
+    }
+
+    if (!serializedProofRequest) {
+      throw new Error("Proof quest field cannot be empty!");
+    }
+
+    const idPOD = POD.deserialize(serializedIDPOD);
+    const paytsubPOD = POD.deserialize(serializedPaystubPOD);
+
+    const proofRequest = JSON.parse(serializedProofRequest);
+    // https://docs.pcd.team/functions/_pcd_gpc.deserializeGPCProofConfig.html
+    const proofConfig = deserializeGPCProofConfig(proofRequest.proofConfig);
+    const membershipLists = proofRequest.membershipLists;
+    const externalNullifier = proofRequest.externalNullifier || "ZooKyc";
```

Comment:
Watermark and external nullifier are allowed to be undefined.  Is there logic behind why you want to set default values?

Reply:
Not really, removing the default values.